tensorboard/uploader/proto/export_service.proto

```diff
@@ -133,4 +144,33 @@ message StreamExperimentDataResponse {
     // Value of the point at this step / timestamp.
     repeated .tensorboard.TensorProto values = 3;
   }
+
+  message BlobSequencePoints {
+    // Step index within the run.
+    repeated int64 steps = 1;
+    // Timestamp of the creation of this point.
+    repeated google.protobuf.Timestamp wall_times = 2;
+    // Value of the BLOB sequence at this step / timestamp.
+    repeated BlobSequence values = 3;
+  }
+}
+
+message StreamBlobRequest {
+  string blob_id = 1;
+}
+
+message StreamBlobResponse {
+  // The bytes in this chunk.
+  bytes data = 1;
+  // The position in the blob where this chunk begins.
+  // This must equal the sum of the sizes of the chunks sent so far.  Ignored
+  // if no data is provided.
+  int64 offset = 2;
+  // TODO(b/150443776): Add `crc32a = 3;` once the server can populate it.
+  reserved 3;
+  // Indicates that this is the last chunk of the stream.
+  bool final_chunk = 4;
+  // CRC32C of the entire blob.  Required, to protect against data corruption.
+  // This should be set only when finalize_object=True.
+  fixed32 final_crc32c = 5;
```

Comment:
This is a bit confusing; it can’t be both required and only set
sometimes. Given that in proto3 it’s not possible to distinguish between
a `fixed32` of value 0 and a missing `fixed32`, perhaps what’s actually
meant here is that when `final_chunk` is `true`, this always represents
a CRC-32 (even if its value happens to be `0`), and that otherwise the
value is not expected to be set and should be ignored?

(Also, presumably `finalize_object` is meant to be `final_chunk`?)

Reply:
This is copied from an internal equivalent proto. I've fixed the doc-string issues you raised. The corresponding issues in the internal origin of this code will go away once this is sync'ed and the internal code is deduplicated in favor of this one.